packages/editor/src/client/routes/pageEditor.tsx

```diff
@@ -347,6 +347,7 @@ export function PageEditor({id}: PageEditorProps) {
 
       <Modal show={isPublishDialogOpen} size={'sm'} onHide={() => setPublishDialogOpen(false)}>
         <PublishPagePanel
+          url={pageData?.page?.draft?.url || pageData?.page?.latest?.url}
```

Comment:
i suggest you put the url into the `metadata`. Then you'll always have the correct version when displaying it here.